api/src/main/java/org/apache/iceberg/transforms/Transforms.java

```diff
@@ -94,14 +110,18 @@ private Transforms() {
    */
   @SuppressWarnings("unchecked")
   public static <T> Transform<T, Integer> year(Type type) {
+    return year(type, ZoneOffset.UTC);
+  }
+
+  public static <T> Transform<T, Integer> year(Type type, ZoneOffset zoneOffset) {
```

Comment:
The cases for `date` need to assert that the offset is either null or UTC. I don't think it makes sense for a date to use the offset, so if there is an offset it should be rejected and not ignored.